src/main/java/pl/connectis/Spotify_API_CLI/authorization/AuthorizationCodeAuthorization.java

```diff
@@ -0,0 +1,127 @@
+package pl.connectis.Spotify_API_CLI.authorization;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.ApplicationContext;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Component;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import java.io.IOException;
+import java.util.Base64;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.ReentrantLock;
+
+@Slf4j
+@Component("authorization_code")
+public class AuthorizationCodeAuthorization {
+
+    private final AuthorizationCode authorizationCode;
+    private final ApplicationContext context;
+    private final RestTemplate restTemplate;
+    private final HttpHeaders httpHeaders;
+    private final ReentrantLock lock;
+    private final Condition condition;
+    @Value("${authorizationMethod}")
+    private String authorizationMethod;
+    @Value("${clientId}")
+    private String clientID;
+    @Value("${clientSecret}")
+    private String clientSecret;
+    @Value("${userAuthorizationUri}")
+    private String userAuthorizationUri;
+    @Value("${accessTokenUri}")
+    private String accessTokenUri;
+    @Value("{$spring.main.web-application-type}")
+    private String webApplicationType;
```

Comment:
Opakuj to w klase z annotacja `@ConfigurationProperties`

Reply:
ok